### PR TITLE
Markdown linting

### PR DESCRIPTION
**Pull request by josecelano**
I've added a new workflow to check markdown files lint. I've added the same default configuration as the Visual Studio Code plugin that we are using: https://github.com/DavidAnson/vscode-markdownlint/blob/main/.markdownlint.json

### CODE REVIEW

**Comment by josecelano** [comment]
If you want to run it locally you can:

1.- Use `docker`

```
docker run --rm \
    -v "$(pwd):/app:ro" \
    avtodev/markdown-lint:v1 \
    --config /app/.markdownlint.json \
    /**/*.md
```

2. Install `markdown-cli`

https://github.com/igorshubovych/markdownlint-cli

```
markdownlint '**/*.md'     <- list error
markdownlint -f '**/*.md'  <- auto fix
```

3. Install Visual Studio Code plugin

https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint

**Comment by josecelano** [comment]
We have 29 errors:

[lint-errors.txt](https://github.com/Nautilus-Cyberneering/chinese-ideographs/files/7526969/lint-errors.txt)

We have to fix them before merging this PR:

File: .markdownlint.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+			"Markdown",
+			"markdown-it",
+			"markdownlint",
+			"Node.js",
+			"Shift",
+			"Visual Studio Code"
+		],
+		"code_blocks": false
+	}
+}
```
Comment (cgbosse) [inline .markdownlint.json:40]: No Idea what this means.
Does one have to add a new line?

File: contrib/github-merge/README.md [hunk 2]
```diff
@@ -52,11 +53,11 @@ limit for authenticated requests is much higher. If you start running
 into rate limiting errors it can be useful to set an authentication token
```
Comment (cgbosse) [inline contrib/github-merge/README.md:53]: This sentences seems off some way if you read the following.

**Review by cgbosse — APPROVED** [review]
Overall the changes are ok.
However in the licensing markdown files I did leave them mostly as they were.
Since it stated that we could not change them.

File: .markdownlint.json (hunk 1 above)
Comment (josecelano) [reply .markdownlint.json:40]: You have the details for all rules here: https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md044---proper-names-should-have-the-correct-capitalization. Anyway, there is another PR: https://github.com/Nautilus-Cyberneering/chinese-ideographs/pull/105 where I was trying GitHub super-linter. That linter also includes markdown linter. Maybe we should use only one, the other one and discard this PR.

**Comment by josecelano** [comment]
I've just talked to @cgbosse , we are going to fix the warnings and merge this PR but in the future we will use the generic GitHub linter that we have added on the other PR: https://github.com/Nautilus-Cyberneering/chinese-ideographs/pull/105. There are a lot of errors, we have to find out how to auto fix them and fix manually the ones that are not fixed automatically.

**Comment by josecelano** [comment]
[errors_20211124.txt](https://github.com/Nautilus-Cyberneering/chinese-ideographs/files/7594903/errors_20211124.txt)

Updated list of errors:
```
(base) josecelano@josecelano:~/Documents/github/Nautilus-Cyberneering/chinese-ideographs$ markdownlint '**/*.md'
documentation/Contributor_Code_of_Conduct.md:47 MD003/heading-style/header-style Heading style [Expected: atx; Actual: setext]
documentation/Contributor_Code_of_Conduct.md:47 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "<<<<<<< Updated upstream"]
documentation/Running_GitHub_Actions_Locally.md:11 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
documentation/Running_GitHub_Actions_Locally.md:22 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
documentation/Running_GitHub_Actions_Locally.md:33 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
licensing/cc-by-sa.md:51 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "## Section 2 – Scope"]
licensing/cc-by-sa.md:93 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "## Section 3 – License Conditi..."]
licensing/cc-by-sa.md:139 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "## Section 4 – Sui Generis Dat..."]
licensing/cc-by-sa.md:159 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "## Section 5 – Disclaimer of W..."]
licensing/cc-by-sa.md:175 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "## Section 6 – Term and Termin..."]
licensing/cc-by-sa.md:199 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "## Section 7 – Other Terms and..."]
licensing/cc-by-sa.md:213 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "## Section 8 – Interpretation"]
```

**Pull request merged**